apps/api/src/stripe/events/stripe-payment.service.ts

```diff
@@ -329,9 +332,10 @@ export class StripePaymentService {
   @StripeWebhookHandler('invoice.paid')
   async handleInvoicePaid(event: Stripe.Event) {
     const invoice: Stripe.Invoice = event.data.object as Stripe.Invoice
-    Logger.log('[ handleInvoicePaid ]', invoice)
 
-    let metadata: DonationMetadata = {
+    Logger.log('[ handleInvoicePaid ]', invoice)
+    const charge = await this.stripeService.findChargeById(invoice.charge as string)
```

Comment:
We are using this variable 30 lines below.
Can we move this line closer to:

`    const paymentData = getInvoiceData(invoice, charge)`